src/main/java/com/feelmycode/parabole/service/UserService.java

```diff
@@ -41,19 +41,19 @@ public User signup(@NotNull UserSignupDto dto) {
             dto.toEntity(dto.getEmail(), dto.getUsername(), dto.getNickname(), dto.getPhone(), dto.getPassword()));
     }
 
-    public boolean signin(@NotNull UserSigninDto dto) {
+    public User signin(@NotNull UserSigninDto dto) {
         log.info("email: {}, password: {}", dto.getEmail(), dto.getPassword());
         if (dto.getEmail().equals("") || dto.getPassword().equals("")) {
             throw new ParaboleException(HttpStatus.BAD_REQUEST, "로그인 입력란에 채우지 않은 란이 있습니다.");
         }
-        if (userRepository.findByEmail(dto.getEmail()) == null) {
+        User user = userRepository.findByEmail(dto.getEmail());
+        if (user == null) {
             throw new ParaboleException(HttpStatus.BAD_REQUEST, "입력하신 이메일을 가진 사용자가 존재하지 않습니다");
         }
-        if (userRepository.findByEmail(dto.getEmail()) != null
-            && !userRepository.findByEmail(dto.getEmail()).getPassword().equals(dto.getPassword())) {
+        if (user != null && !user.getPassword().equals(dto.getPassword())) {
```

Comment:
```suggestion
        if (!user.getPassword().equals(dto.getPassword())) {
```
user != null 이 이미 걸러진거 같아요! 지워도 괜찮을거 같아요

Reply:
반영완